source/Playnite/Database/GameDatabase.cs

```diff
@@ -1059,34 +1059,60 @@ public List<Game> ImportGames(LibraryPlugin library, bool forcePlayTimeSync, ILi
                     }
                     else
                     {
-                        existingGame.IsInstalled = newGame.IsInstalled;
-                        existingGame.InstallDirectory = newGame.InstallDirectory;
+                        var existingGameUpdated = false;
+                        if (existingGame.IsInstalled != newGame.IsInstalled)
+                        {
+                            existingGame.IsInstalled = newGame.IsInstalled;
+                            existingGameUpdated = true;
+                        }
+                        
+                        if (existingGame.InstallDirectory.Equals(newGame.InstallDirectory, StringComparison.OrdinalIgnoreCase) == false)
+                        {
+                            existingGame.InstallDirectory = newGame.InstallDirectory;
+                            existingGameUpdated = true;
+                        }
+                        
                         if (existingGame.PlayAction == null || existingGame.PlayAction.IsHandledByPlugin)
                         {
-                            existingGame.PlayAction = newGame.PlayAction;
+                            if (existingGame.PlayAction != newGame.PlayAction)

```

Comment:
GameAction object doesn't implement != operator, so this check will not work and always will be false.